src/cryptonote_core/blockchain.cpp

```diff
@@ -438,8 +435,15 @@ bool Blockchain::init(BlockchainDB* db, const network_type nettype, bool offline
   m_async_pool.create_thread(boost::bind(&boost::asio::io_service::run, &m_async_service));
 
 #if defined(PER_BLOCK_CHECKPOINT)
-  if (m_nettype != FAKECHAIN)
-    load_compiled_in_block_hashes();
+  if (m_fast_sync && get_checkpoints != nullptr && m_nettype != FAKECHAIN)
+  {
+    size_t checkpoints_size;
+    const unsigned char *checkpoints_data = get_checkpoints(m_nettype, checkpoints_size);
+    if (checkpoints_data != nullptr && checkpoints_size > 0)
+    {
+      load_compiled_in_block_hashes(checkpoints_data, checkpoints_size);
+    }    
+  }
```

Comment:
That (and below) looks like lots of unneeded changes moving stuff around for no clear reason. Why ?

Reply:
Fixed